src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rest/QueryBatchRequestHandler.java

```diff
@@ -0,0 +1,309 @@
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.StatExceptionCode;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.StatsCollector;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.config.PluginSettings;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsRestUtil;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.model.MetricsModel;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.net.NetClient;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ReaderMetricsProcessor;
+import com.google.common.annotations.VisibleForTesting;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.security.InvalidParameterException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.apache.logging.log4j.util.Supplier;
+import org.jooq.Record;
+import org.jooq.Result;
+
+/**
+ * Request handler that supports querying batch metrics from an EC2 instance
+ *
+ * <p>Return 1 minute of CPU_Utilization metrics sampled at a 5s sampling period:
+ * "http://localhost:9600/_opendistro/_performanceanalyzer/batch?metrics=CPU_Utilization&starttime=1566413975000&endtime=1566413980000"
+ *
+ * <p>Return 1 minute of CPU_Utilization and Latency metrics sampled at a 10s sampling period:
+ * "http://localhost:9600/_opendistro/_performanceanalyzer/batch?metrics=CPU_Utilization,Latency&starttime=1566413975000&endtime=1566413980000&samplingperiod=10"
+ *
+ * <p>Return format:
+ * {
+ *   "1594412650000": {
+ *     "CPU_Utilization": {
+ *       "fields": [
+ *         {
+ *           "name: "IndexName",
+ *           "type": "VARCHAR"
+ *         },
+ *         <...>
+ *       ]
+ *       "records": [
+ *         [
+ *           "pmc",
+ *           <...>
+ *         ],
+ *         <...>
+ *       ]
+ *     }
+ *   }
+ * }
+ */
+public class QueryBatchRequestHandler extends MetricsHandler implements HttpHandler {
+
+  private static final Logger LOG = LogManager.getLogger(QueryBatchRequestHandler.class);
+
+  private static final int TIME_OUT_VALUE = 2;
+  private static final TimeUnit TIME_OUT_UNIT = TimeUnit.SECONDS;
+  private NetClient netClient;
+  MetricsRestUtil metricsRestUtil;
+
+  public static final int DEFAULT_MAX_DATAPOINTS = 100800;  // Must be non-negative
+  public static final long DEFAULT_SAMPLING_PERIOD_MILLIS = 5000;  // Must be a multiple of 5000
```

Comment:
Let's also add the reason for the numbers here ?

Reply:
Max datapoints was capped to prevent excessive memory consumption. Will note that in README and design docs. However, I don't see the need to note anything about DEFAULT_SAMPLING_PERIOD_MILLIS.